src/base/sexpr.zig

```diff
@@ -0,0 +1,218 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const testing = std.testing;
+
+/// Represents a node in an S-expression tree.
+pub const Node = union(enum) {
```

Comment:
nit: maybe name this `Expr` then an `Expr` can be a `node` or `string` or etc.

Reply:
Leads to `called appendChildUnsafe on a Node that is not a node` becoming `called appendChildUnsafe on a Expr that is not a node, it is a

Reply:
Done in 58efb0416a